force-app/main/default/classes/RD2_ChangeHistoryService.cls

```diff
@@ -0,0 +1,136 @@
+/*
+    Copyright (c) 2021 Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @group Recurring Donations
+* @description All services specific to Recurring Donation Change History
+*
+*/
+public without sharing class RD2_ChangeHistoryService {
```

Comment:
I think this specific class should be set to either `with sharing` or `inherited sharing`. What's needed though, is the ability to instantiate and insert data into an object the User doesn't have permission to. At the moment, I'm pretty sure this is only executed in a trigger context - which means it's already executing in System permissions (which is all the more reason to stick with `inherited sharing`). 

It is very possible that at some point this class may be called from another non-trigger entry point. In that case, the sharing context does become more important. My recommendation is to use an inner class in this class explicitly defined as `without sharing` that isolates the operations that will always required escalated perms. This way everything else runs _normally_ and only the very specific operations that need the extra perms always run in that mode. Plus, it helps self-document the operations and reasoning so that this class doesn't get called out by the inevitable security review.

Reply:
This is a great approach, I'll work on this today. I'll also write a unit test leveraging the Standard User profile to make sure this works as expected.